### PR TITLE
[BEAM-439] Remove optional args from start/finish bundle methods

### PR DESCRIPTION
**Pull request by silviulica**
@robertwb can you please take a look?
Technically we do not want to allow deferred side inputs but it is much simple conceptually to remove
extra arguments altogether.

### CODE REVIEW

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -895,6 +903,7 @@ def apply(self, pcoll):
       key_type, _ = input_type.tuple_types
 
     runtime_type_check = (
+        pcoll.pipeline.options is not None and
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: We should't have to do this check anymore.

File: sdks/python/apache_beam/transforms/core.py [hunk 2]
```diff
@@ -857,7 +863,7 @@ def typed(transform):
 
 
 @ptransform_fn
-def CombinePerKey(pcoll, fn, *args, **kwargs):  # pylint: disable=invalid-name
+def CombinePerKey(label, pcoll, fn, *args, **kwargs):  # pylint: disable=invalid-name
```
Comment (robertwb) [inline sdks/python/apache_beam/transforms/core.py]: Bad merge?

**Comment by robertwb** [comment]
FWIW, I think removing all side inputs as you've done is actually cleaner, as whether a side input is deferred or known at runtime should not be detectable by the caller (and allows the callee more freedom). If something is needed by start/finish, it can be placed in the constructor.

**Comment by silviulica** [comment]
Bad merge! Let me clean this up.

**Comment by silviulica** [comment]
Abandoning this PR and starting clean to not have some messed up commit history.

**Pull request closed without merging**